frontend/lib/src/models/transaction.dart

```diff
@@ -11,37 +12,66 @@ enum TransactionTypes { CREDIT, DEPT, TRANSFER }
 enum Category { SAVINGS, SALARY, BILLS, SENDER, RECIPIENT }
 
 class Transaction {
-  String? name;
-  String? date;
-  String amount;
-  String description;
+  String? id;
+  DateTime? transactionDate;
   TransactionTypes transactionType;
   Category category;
+  double amount;
+  String description;
+  String? accountName;
+  String? senderName;
+  String? receiverName;
 
   Transaction({
-    this.name,
-    this.date,
-    required this.amount,
-    required this.description,
+    this.id,
```

Comment:
shouldnt id and transaction date be required here?

Reply:
postTransaction doesn't have any id or dates yet

Reply:
were talking about the model here